src/testdir/test_quickfix.vim

```diff
@@ -6496,4 +6496,70 @@ func Test_efm_format_b()
   call setqflist([], 'f')
 endfunc
 
+func XbufferTests_range(cchar)
+  call s:setup_commands(a:cchar)
+
+  enew!
+  let lines =<< trim END
+    Xtestfile7:700:10:Line 700
+    Xtestfile8:800:15:Line 800
+  END
+  silent! call setline(1, lines)
+  norm! Vy
+  if (a:cchar == 'c')
+     exe "'<,'>cbuffer!"
+  else
+    exe "'<,'>lbuffer!"
+  endif
+  let l = g:Xgetlist()
+  call assert_true(len(l) == 1 &&
+	\ l[0].lnum == 700 && l[0].col == 10 && l[0].text ==# 'Line 700')
+
+  enew!
+  let lines =<< trim END
+    Xtestfile9:900:55:Line 900
+    Xtestfile10:950:66:Line 950
+  END
+  silent! call setline(1, lines)
+  if (a:cchar == 'c')
```

Comment:
Instead of this if-else, can you use `Xgetbuffer` here?

Reply:
I initially used `Xbuffer` with `command! -nargs=* -bang -range=% Xbuffer <mods><line1>,<line2>cbuffer<bang> <args>` and then using `Xbuffer` around line '6510' , but unfortunately, then it didn't fail properly, so I resorted back to using the raw `:c/l` commands for testing. Also here on this line.